pseudomonas_analysis/0_prepare_reference_gene_file.ipynb

```diff
@@ -0,0 +1,2456 @@
+       "ig_68616_69272_at                 NaN            NaN            NaN  \n",
+       "ig_69272_68616_at                 NaN            NaN            NaN  \n",
+       "ig_721556_727608_s_at             NaN            NaN            NaN  \n",
+       "ig_727608_721556_s_at             NaN            NaN            NaN  \n",
+       "ig_773696_774416_at               NaN            NaN            NaN  \n",
+       "ig_774416_773696_at               NaN            NaN            NaN  \n",
+       "ig_785174_785969_at               NaN            NaN            NaN  \n",
+       "ig_785969_785174_at               NaN            NaN            NaN  \n",
+       "ig_788253_789144_at               NaN            NaN            NaN  \n",
+       "ig_789144_788253_at               NaN            NaN            NaN  \n",
+       "ig_863300_864095_at               NaN            NaN            NaN  \n",
+       "ig_864095_863300_at               NaN            NaN            NaN  \n",
+       "ig_884172_884799_at               NaN            NaN            NaN  \n",
+       "ig_884799_884172_at               NaN            NaN            NaN  \n",
+       "ig_901046_901934_at               NaN            NaN            NaN  \n",
+       "ig_901934_901046_at               NaN            NaN            NaN  \n",
+       "ig_991198_991830_at               NaN            NaN            NaN  \n",
+       "ig_991830_991198_at               NaN            NaN            NaN  \n",
+       "\n",
+       "[5900 rows x 73 columns]"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# For each experiment get df\n",
+    "# For each df, if label gene as DEGs based on log2FC>1 and FDR<0.05\n",
+    "# Concatenate series\n",
+    "num_experiments = len(reference_Robject)\n",
+    "reference_stats_df = pd.DataFrame()\n",
+    "\n",
+    "for i in range(num_experiments):\n",
+    "    # Get df for experiment\n",
+    "    reference_df = pandas2ri.ri2py(reference_Robject[i])\n",
+    "    \n",
+    "    reference_df = reference_df.set_index(\"ProbeID\")\n",
+    "    \n",
+    "    # Find DEGs\n",
+    "    degs_ids = list(reference_df[(abs(reference_df[\"Log2FC\"])>1) & (reference_df[\"FDR\"]<0.05)].index)\n",
+    "    reference_df[\"DEG\"] = 0\n",
+    "    reference_df.loc[degs_ids, \"DEG\"] = 1\n",
+    "    \n",
+    "    # Create df with labels for if gene is DE or not\n",
+    "    if i == 0:\n",
+    "        reference_stats_df = reference_df[\"DEG\"].to_frame(\"experiment_0\") \n",
+    "    else:\n",
+    "        reference_stats_df = pd.merge(\n",
+    "            reference_stats_df,\n",
+    "            reference_df[\"DEG\"].to_frame(f\"experiment_{i}\"),\n",
+    "            left_index=True, \n",
+    "            right_index=True,\n",
+    "            how='left'\n",
+    "        )\n",
```

Comment:
If this cell runs slowly, you can probably speed it up by keeping track of the DEGs for each experiments in a dictionary then add create the dataframe at the end

Reply:
This cell is very quick but thank you for the nice tip. Always helpful to know how to cleanup my code